searx/network/client.py

```diff
@@ -220,18 +577,5 @@ def init():
     ):
         logging.getLogger(logger_name).setLevel(logging.WARNING)
 
-    # loop
-    def loop_thread():
-        global LOOP
-        LOOP = asyncio.new_event_loop()
-        LOOP.run_forever()
-
-    thread = threading.Thread(
-        target=loop_thread,
-        name='asyncio_loop',
-        daemon=True,
-    )
-    thread.start()
-
 
 init()
```

Comment:
Isn't it better the init (set log-levels) is done by the application? .. set log-level on import of a module can confuse, when other modules do the same (I know that is not the case in SearXNG, but AFIAK know its a anti-pattern to set log-levels or do other side-effects that are not related directly to the module while importing the module).

Reply:
> ..  searx/__init__.py because this is where the logging initialization is done

Sadly that is true .. IMO its the same issue: *"set log-levels or do other side-effects that are not related directly to the module while importing the module"*

https://github.com/searxng/searxng/blob/24561370b7702846124ab083701792deee310c5b/searx/__init__.py#L91-L96

> what do you think?

I think all initialization should be done when the application context is build up.  We do not have an *application factory*, we build the application while import `searx.webapp`.

To limit the damage, we should perform initializations only when importing the `searx.webapp` (or where needed, during the tests).

https://github.com/searxng/searxng/blob/24561370b7702846124ab083701792deee310c5b/searx/webapp.py#L1318-L1325

Reply:
> Do you have time to add the change in your branch?

I will give it a try .. but will take a while / first I want to read, format and .. **understand** the doc-strings :)